invenio_app_ils/ill/api.py

```diff
@@ -88,11 +91,57 @@ def delete(self, **kwargs):
 )
 
 
+class IllValidator(RecordValidator):
+    """Ill record validator."""
+
+    def validate_cancel(self, status, cancel_reason):
+        """Validate decline is correct."""
+        if status == "CANCELLED" and not cancel_reason:
+            raise ILLError(
+                "You have to provide a cancel reason when cancelling a request"
+            )
+        if cancel_reason and not status == "CANCELLED":
+            raise ILLError(
+                "If you select a cancel reason you need to select"
+                " \"Cancelled\" in the state"
+            )
+
+    def ensure_document_exists(self, document_pid):
+        """Ensure document exists or raise."""
+        from invenio_app_ils.documents.api import Document
```

Comment:
Please do not hardcode classes import whenever possible. In this case it is not very relevant, but we tend to copy/paste around...
```suggestion
        Document = current_app_ils.document_record_cls
```